flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvRowDataDeserializationSchema.java

```diff
@@ -0,0 +1,458 @@
+				return this::convertToBoolean;
+			case TINYINT:
+				return jsonNode -> Byte.parseByte(jsonNode.asText().trim());
+			case SMALLINT:
+				return jsonNode -> Short.parseShort(jsonNode.asText().trim());
+			case INTEGER:
+			case INTERVAL_YEAR_MONTH:
+				return this::convertToInt;
+			case BIGINT:
+			case INTERVAL_DAY_TIME:
+				return this::convertToLong;
+			case DATE:
+				return this::convertToDate;
+			case TIME_WITHOUT_TIME_ZONE:
+				return this::convertToTime;
+			case TIMESTAMP_WITH_TIME_ZONE:
+			case TIMESTAMP_WITHOUT_TIME_ZONE:
+				return this::convertToTimestamp;
+			case FLOAT:
+				return this::convertToFloat;
+			case DOUBLE:
+				return this::convertToDouble;
+			case CHAR:
+			case VARCHAR:
+				return this::convertToString;
+			case BINARY:
+			case VARBINARY:
+				return this::convertToBytes;
+			case DECIMAL:
+				return createDecimalConverter((DecimalType) type);
+			case ARRAY:
+				return createArrayConverter((ArrayType) type);
+			case ROW:
+				return createRowConverter((RowType) type, false);
+			case MAP:
+			case MULTISET:
+			case RAW:
+			default:
+				throw new UnsupportedOperationException("Unsupported type: " + type);
+		}
+	}
+
+	private boolean convertToBoolean(JsonNode jsonNode) {
+		if (jsonNode.isBoolean()) {
+			// avoid redundant toString and parseBoolean, for better performance
+			return jsonNode.asBoolean();
+		} else {
+			return Boolean.parseBoolean(jsonNode.asText().trim());
+		}
+	}
+
+	private int convertToInt(JsonNode jsonNode) {
+		if (jsonNode.canConvertToInt()) {
+			// avoid redundant toString and parseInt, for better performance
+			return jsonNode.asInt();
+		} else {
+			return Integer.parseInt(jsonNode.asText().trim());
+		}
+	}
+
+	private long convertToLong(JsonNode jsonNode) {
+		if (jsonNode.canConvertToLong()) {
+			// avoid redundant toString and parseLong, for better performance
+			return jsonNode.asLong();
+		} else {
+			return Long.parseLong(jsonNode.asText().trim());
+		}
+	}
+
+	private double convertToDouble(JsonNode jsonNode) {
+		if (jsonNode.isDouble()) {
+			// avoid redundant toString and parseDouble, for better performance
+			return jsonNode.asDouble();
+		} else {
+			return Double.parseDouble(jsonNode.asText().trim());
+		}
+	}
+
+	private float convertToFloat(JsonNode jsonNode) {
+		if (jsonNode.isDouble()) {
+			// avoid redundant toString and parseDouble, for better performance
+			return (float) jsonNode.asDouble();
+		} else {
+			return Float.parseFloat(jsonNode.asText().trim());
+		}
+	}
+
+	private int convertToDate(JsonNode jsonNode) {
+		// csv currently is using Date.valueOf() to parse date string
+		return (int) Date.valueOf(jsonNode.asText()).toLocalDate().toEpochDay();
+	}
+
+	private int convertToTime(JsonNode jsonNode) {
+		// csv currently is using Time.valueOf() to parse time string
+		LocalTime localTime = Time.valueOf(jsonNode.asText()).toLocalTime();
+		// get number of milliseconds of the day
+		return localTime.toSecondOfDay() * 1000;
```

Comment:
Loose mills precision here, use `toNanoOfDay`?
BTW, add tests for this.

Reply:
This is also align with previous behavior. `Time.valueOf()` doesn't support to parse milli/nano seconds.

Reply:
Previous behavior is because old planner only support seconds, but blink support mills and more.
We are using `LocalTime` instead of `Time` now. If you don't want change it you should add a TODO and create a JIRA.

Reply:
Thanks @JingsongLi , I created FLINK-17525 to track this.